scripts/update-all

```diff
@@ -52,7 +52,9 @@ function run()
     node "./scripts/run.js" "./github" "w3c.json" &&
     node "./scripts/run.js" "./github" "wicg.json" &&
     node "./scripts/run.js" "./github" "biblio.json" &&
-    node "./scripts/run.js" "./github" "whatwg.json";
+    node "./scripts/run.js" "./github" "whatwg.json" &&
+    node "./scripts/scripts/css-alias.js" &&
+    node "./scripts/list-refs.js"; # regenerate, as CSS aliases may have changed
```

Comment:
Why are you regenerating that list here? This seems like a pretty big smell.

Reply:
Yeah, you really can't do that. You're basically by-passing the whole check to verify that you're not removing entries by running this at this point.

Instead you need to figure out what entries are getting removed and add them automatically.

Reply:
If, for example: 

 * today "css-foo" references "css-foo-1", 
 * tomorrow  "css-foo" becomes "css-foo-2"

Now all the "css-foo-YYYYMMDD" references need to be updated... this is where is gets complicated. Perhaps "date citing" an an alias shouldn't be allowed. Or, we need to figure out how to retain these old references as more aliases.

Reply:
Absolutely. I now remember why trying to fix this index issue in the past had made me sad.

The correct way of solving the problem is stacking everything onto "foo" including version numbers ("foo-1"), rec track status ("foo-1-CR"), and dated releases ("foo-YYYYMMDD") like so:

```json
{
    "css-grid": {
        "versions": {
            "1": {...},
            "2": {...},
            "20110407": {...},
            "20120322": {...},
            "20121106": {...},
            "20130402": {...},
            "20130910": {...},
            "20140123": {...},
            "20140513": {...},
            "20150317": {...},
            "20150806": {...},
            "20150917": {...},
            "20160519": {...},
            "20160929": {...},
            "20170209": {...},
            "20170509": {...},
            "20171214": {...},
            "20180206": {...},
            "20180427": {...},
            "20180628": {...},
            "20180804": {...}
        }
    }
}
```

Compared to what we have which is more:

```json
{
    "css-grid-1": {
        "versions": {
            "20110407": {...},
            "20120322": {...},
            "20121106": {...},
            "20130402": {...},
            "20130910": {...},
            "20140123": {...},
            "20140513": {...},
            "20150317": {...},
            "20150806": {...},
            "20150917": {...},
            "20160519": {...},
            "20160929": {...},
            "20170209": {...},
            "20170509": {...},
            "20171214": {...}
        }
    },
    "css-grid-2": {
        "versions": {
            "20180206": {...},
            "20180427": {...},
            "20180628": {...},
            "20180804": {...}
        },
    }
}
```

This is actually hard to do given the existing data set.

Reply:
Well, I guess even that has some limitations if there are two concurrent versions releasing new drafts on the same day. :-/

Reply:
Maybe your best bet is to run this manually from time to time for now?

Reply:
😭... indeed. ok... argh. Ok, it's doable. Need to find all "css-grid-*" can then clone all `versions` in. Will try tomorrow.

Reply:
It might be worth merging this minus the changes to update-all, and really giving the data model some thought before committing to something else, there.

Reply:
Yeah, having a dated reference to the unleveled ref doesn't make sense; either it only resolves to a particular version's dates (which then become invalid when a new version comes out and the unleveled ref moves) or it resolves to all of them (and then you can't disambiguate when two versions of a spec come out on the same day).  The latter issue is *definitely* possible, especially now that we have such easier publishing via Echidna; I wouldn't be surprised if it had already happened, actually.

Reply:
> Yeah, having a dated reference to the unleveled ref doesn't make sense; 

Agree.

> The latter issue is definitely possible, especially now that we have such easier publishing via Echidna; I wouldn't be surprised if it had already happened, actually.

I has... I spotted one already.

Reply:
Alright, so we'll have to maintain the existing ones for backwards compat and start anew with the rest. Maybe worth looking into the JSON API while we're at it?